experimental/telemetry/tracker.go

```diff
@@ -66,20 +97,41 @@ func NewTracker(
 		serverVersion:      serverVersion,
 		pluginID:           pluginID,
 		pluginVersion:      pluginVersion,
-		enabled:            enableDiagnostics,
+		enabledLogging:     config.EnabledLogging,
+		enabledTracking:    config.EnabledTracking,
+		logger:             l,
 	}
 }
 
+func (t *tracker) ReloadConfig(config TrackerConfig) {
+	if config.EnabledTracking != t.enabledTracking {
+		if config.EnabledTracking {
+			t.debugf("Enabling plugin telemetry")
+		} else {
+			t.debugf("Disabling plugin telemetry")
+		}
+		t.enabledTracking = config.EnabledTracking
+	}
+	t.enabledLogging = config.EnabledLogging
```

Comment:
Same 0/5 preference of newlines after blocks that have code in the same block scope

```suggestion
		}

		t.enabledTracking = config.EnabledTracking
	}

	t.enabledLogging = config.EnabledLogging
```